onnx_xla/onnx_xla_interface.cc

```diff
@@ -45,108 +61,118 @@ struct BackendControl {
 //TODO: Determining # of CPU, GPU, TPU devices and return
 ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI ONNXIFI_SYMBOL_NAME(
     onnxGetBackendIDs)(onnxBackendID *backendIDs, size_t *numBackends) {
-  try {
-    *backendIDs = (onnxBackendID)(new OnnxXlaBackendID());
+  ONNXIFI_TRY_CATCH(([&, backendIDs, numBackends] {
+    *numBackends = 0;
+    *backendIDs = reinterpret_cast<onnxBackendID>(new OnnxXlaBackendID());
     *numBackends = 1;
     return ONNXIFI_STATUS_SUCCESS;
-  }
-  ONNXIFI_CATCH_EXCPETION();
+  }))
 }
 
 //Free memory for given backend ID
 ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI
 ONNXIFI_SYMBOL_NAME(onnxReleaseBackendID)(onnxBackendID backendID) {
-  try {
-    auto *backend_id = reinterpret_cast<OnnxXlaBackendID *>(backendID);
-    if (!backend_id) {
+  ONNXIFI_TRY_CATCH(([&, backendID] {
```

Comment:
No need to capture `backendID` explicitly as you are doing`&` already.

Reply:
fixed